app/deps/libusb.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -ex
+DEPS_DIR=$(dirname ${BASH_SOURCE[0]})
+cd "$DEPS_DIR"
+. common
+
+VERSION=1.0.27
+FILENAME=libusb-$VERSION.tar.gz
+PROJECT_DIR=libusb-$VERSION
+SHA256SUM=e8f18a7a36ecbb11fb820bd71540350d8f61bcd9db0d2e8c18a6fb80b214a3de
+
+cd "$SOURCES_DIR"
+
+if [[ -d "$PROJECT_DIR" ]]
+then
+    echo "$PWD/$PROJECT_DIR" found
+else
+    get_file "https://github.com/libusb/libusb/archive/refs/tags/v$VERSION.tar.gz" "$FILENAME" "$SHA256SUM"
```

Comment:
The source release is the .bz2 file, also available on SourceForge https://sourceforge.net/projects/libusb/files/libusb-1.0/

Reply:
Thank you for your review :+1: I changed to use the `.tar.bz2`:

```diff
diff --git a/app/deps/libusb.sh b/app/deps/libusb.sh
index 26f0140ba..97fc3c72f 100755
--- a/app/deps/libusb.sh
+++ b/app/deps/libusb.sh
@@ -5,9 +5,9 @@ cd "$DEPS_DIR"
 . common
 
 VERSION=1.0.27
-FILENAME=libusb-$VERSION.tar.gz
+FILENAME=libusb-$VERSION.tar.bz2
 PROJECT_DIR=libusb-$VERSION
-SHA256SUM=e8f18a7a36ecbb11fb820bd71540350d8f61bcd9db0d2e8c18a6fb80b214a3de
+SHA256SUM=ffaa41d741a8a3bee244ac8e54a72ea05bf2879663c098c82fc5757853441575
 
 cd "$SOURCES_DIR"
 
@@ -15,7 +15,7 @@ if [[ -d "$PROJECT_DIR" ]]
 then
     echo "$PWD/$PROJECT_DIR" found
 else
-    get_file "https://github.com/libusb/libusb/archive/refs/tags/v$VERSION.tar.gz" "$FILENAME" "$SHA256SUM"
+    get_file "https://github.com/libusb/libusb/releases/download/v$VERSION/libusb-$VERSION.tar.bz2" "$FILENAME" "$SHA256SUM"
     tar xf "$FILENAME"  # First level directory is "$PROJECT_DIR"
 fi
 
@@ -33,7 +33,6 @@ else
     mkdir "$HOST"
     cd "$HOST"
 
-    "$SOURCES_DIR/$PROJECT_DIR"/bootstrap.sh
     "$SOURCES_DIR/$PROJECT_DIR"/configure \
         --prefix="$INSTALL_DIR/$HOST" \
         --host="$HOST_TRIPLET" \
```

Why do you release a source different from the git repo (equivalent to the `.tar.gz`)?

Here is the file difference:

<details><summary>diff</summary>

```bash
$ diff -U3 <(cd libusb-clone; tree) <(cd libusb-bz2; tree)
```

```diff
--- /dev/fd/63	2024-03-02 13:26:21.612080700 +0100
+++ /dev/fd/62	2024-03-02 13:26:21.616080749 +0100
@@ -2,14 +2,13 @@
 ├── AUTHORS
 ├── COPYING
 ├── ChangeLog
-├── HACKING
+├── INSTALL
 ├── INSTALL_WIN.txt
 ├── Makefile.am
+├── Makefile.in
 ├── NEWS
 ├── PORTING
 ├── README
-├── README.git
-├── README.md -> README
 ├── TODO
 ├── Xcode
 │   ├── common.xcconfig
@@ -21,6 +20,7 @@
 │   ├── libusb_debug.xcconfig
 │   ├── libusb_release.xcconfig
 │   └── release.xcconfig
+├── aclocal.m4
 ├── android
 │   ├── README
 │   ├── config.h
@@ -33,16 +33,20 @@
 │       ├── examples.mk
 │       ├── libusb.mk
 │       └── tests.mk
-├── appveyor.yml
-├── autogen.sh
-├── bootstrap.sh
+├── compile
+├── config.guess
+├── config.h.in
+├── config.sub
+├── configure
 ├── configure.ac
+├── depcomp
 ├── doc
 │   ├── Makefile.in
 │   ├── doxygen.cfg.in
 │   └── libusb.png
 ├── examples
 │   ├── Makefile.am
+│   ├── Makefile.in
 │   ├── dpfp.c
 │   ├── ezusb.c
 │   ├── ezusb.h
@@ -52,9 +56,11 @@
 │   ├── sam3u_benchmark.c
 │   ├── testlibusb.c
 │   └── xusb.c
+├── install-sh
 ├── libusb
 │   ├── Makefile.am
 │   ├── Makefile.am.extra
+│   ├── Makefile.in
 │   ├── core.c
 │   ├── descriptor.c
 │   ├── hotplug.c
@@ -100,6 +106,14 @@
 │   ├── version.h
 │   └── version_nano.h
 ├── libusb-1.0.pc.in
+├── ltmain.sh
+├── m4
+│   ├── libtool.m4
+│   ├── ltoptions.m4
+│   ├── ltsugar.m4
+│   ├── ltversion.m4
+│   └── lt~obsolete.m4
+├── missing
 ├── msvc
 │   ├── Base.props
 │   ├── Configuration.Application.props
@@ -129,8 +143,10 @@
 │   ├── stress_mt.vcxproj
 │   ├── testlibusb.vcxproj
 │   └── xusb.vcxproj
+├── test-driver
 └── tests
     ├── Makefile.am
+    ├── Makefile.in
     ├── init_context.c
     ├── libusb_testlib.h
     ├── macos.c
@@ -138,10 +154,6 @@
     ├── stress.c
     ├── stress_mt.c
     ├── testlib.c
-    ├── umockdev.c
-    └── webusb-test-shim
-        ├── index.js
-        ├── package-lock.json
-        └── package.json
+    └── umockdev.c
 
-14 directories, 131 files
```

</summary>